console/v1/types_console_quick_start.go

```diff
@@ -0,0 +1,126 @@
+package v1
+
+import (
+	authorizationv1 "k8s.io/api/authorization/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ConsoleQuickStart is an extension for guiding user through various
+// workflows in the OpenShift web console.
+type ConsoleQuickStart struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +kubebuilder:validation:Required
+	// +required
+	Spec ConsoleQuickStartSpec `json:"spec"`
+}
+
+// ConsoleQuickStartSpec is the desired quick start configuration.
+type ConsoleQuickStartSpec struct {
+	// displayName is the display name of the Quick Start.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	DisplayName string `json:"displayName"`
+	// icon is a base64 encoded image that are used as Quick Start's
+	// +optional
+	Icon string `json:"icon,omitempty"`
+	// tags is an array of strings that describe the Quick Start.
+	// +optional
+	Tags []string `json:"tags,omitempty"`
+	// durationMinutes describes approximately how many minutes it will take to complete the Quick Start.
+	// +optional
+	DurationMinutes int `json:"durationMinutes,omitempty"`
+	// description is the description of the Quick Start. (includes markdown)
+	// +optional
+	Description string `json:"description,omitempty"`
+	// prerequisites contains a list of Quick Start names that must be completed before this Quick Start.
+	// +optional
+	Prerequisites []string `json:"prerequisites,omitempty"`
```

Comment:
Do we have a need for prerequisites to be text? Or do we expect prerequisites to always be another quick start reference?
Inquiring with ux on their expectation.

Reply:
@christianvogt I would say both are valid. Therefor string type could handle both.

Reply:
Are you thinking we first try and match a string value against the quick start names in the library? If no match, display as raw text?

Reply:
@jhadvig I don't think we want to overload the field to mean two things. If we want to support both resource references and regular text, I'd make them separate properties.

Reply:
UX said both might be useful.
So perhaps following what Sam said and having two separate fields would be better: `prerequisites string` and `prerequisiteQuickStarts []string`

Reply:
@jhadvig please confirm with PM but from a slack conversation it seems that having a string value is sufficient. Ie arbitrary text and not a reference to another quick start.

Reply:
Changed the description to: `prerequisites contains all prerequisites that need to be met before taking a Quick Start. (includes markdown)`